python/audit/middleware.py

```diff
@@ -40,14 +41,19 @@ def _audit(user: User | AnonymousUser | None = None, username: str = None):
         AuditSession.objects.create(user=user, login_result=result, login_details=details)
 
     def __call__(self, request):
+        try:
+            username = json.loads(request.body.decode("utf-8")).get("username")
+        except json.JSONDecodeError:
```

Comment:
Maybe better to use explcit import `from json.decoder import JSONDecodeError` ?

Reply:
Fixed